### PR TITLE
feat: cleanup event error cases

### PR DESCRIPTION
**Pull request by Stebalien**
1. Fail with LimitExceeded where it makes sense.
2. Validate that we comsume all keys/values.

### CODE REVIEW

**Comment by Stebalien** [comment]
Implements https://github.com/filecoin-project/FIPs/pull/831

**Comment by Stebalien** [comment]
Builds on #1885

File: fvm/src/kernel/default.rs [hunk 1]
```diff
@@ -1079,23 +1078,19 @@ where
                         .into(),
                 );
             }
+
             if header.key_len > MAX_KEY_LEN as u32 {
                 let tmp = header.key_len;
-                return Err(syscall_error!(IllegalArgument; "event key exceeded max size: {} > {MAX_KEY_LEN}", tmp).into());
+                return Err(syscall_error!(LimitExceeded; "event key exceeded max size: {} > {MAX_KEY_LEN}", tmp).into());
```
Comment (fridrik01) [inline fvm/src/kernel/default.rs:1084]: nit, but in https://github.com/filecoin-project/FIPs/pull/831 it lists that we check each entry key individually for valid UTF8  while in our impl we do it all upfront for all keys.

Comment (Stebalien) [reply fvm/src/kernel/default.rs:1084]: Gah, my patch reverts my previous fixes.

Comment (Stebalien) [reply fvm/src/kernel/default.rs:1084]: Fixed.

**Pull request merged**